server/msteams/layer_generators/disconnection_layer.go.tmpl

```diff
@@ -31,7 +31,7 @@ func (c *{{$.Name}}) {{$index}}({{$element.Params | joinParamsWithType}}) {{$ele
 	{{genResultsVars $element.Results false }} := c.Client.{{$index}}({{$element.Params | joinParams}})
 	if err != nil {
 		var graphErr *msteams.GraphAPIError
-		if errors.As(err, &graphErr) && graphErr.StatusCode == http.StatusUnauthorized {
+		if msteams.IsOAuthError(err) || (errors.As(err, &graphErr) && graphErr.StatusCode == http.StatusUnauthorized) {
```

Comment:
Does this mean the current batch of errors when the user's session is invalidated /aren't/ returning `http.StatusUnauthorized` but either no status code or some other code instead? I'm wondering if that means our relying on status unauthorized is unnecessary, or worse: somehow incorrect.

Not blocking, but I'll push a small commit to add logging of the `StatusCode` to this PR so we can get visibility here.